import review_grpc.proto.review_pb2_grpc as pb2_grpc
import review_grpc.proto.review_pb2 as pb2

class ReviewService(pb2_grpc.ReviewServiceServicer):
    # def Hello(self, request, context):
    #     return pb2.HelloReply(message="Xin ch√†o t·ª´ gRPC Django!")
    def CreateReview(self, request, context):
        message = "Response from django"
        success = True
        # üëâ X·ª≠ l√Ω logic ·ªü ƒë√¢y (v√≠ d·ª• ki·ªÉm tra, l∆∞u DB, validate...)
        # ·ªû ƒë√¢y demo ƒë∆°n gi·∫£n: n·∫øu c√≥ email + password th√¨ success = True
        # if email and password:
        #     message = f"ƒê√£ t·∫°o ng∆∞·ªùi d√πng v·ªõi email: {email}"
        #     success = True
        # else:
        #     message = "Thi·∫øu email ho·∫∑c m·∫≠t kh·∫©u"
        #     success = False

        return pb2.ReviewResponse(message=message, success=success)

def grpc_handlers(server):
    pb2_grpc.add_ReviewServiceServicer_to_server(ReviewService(), server)
